programs/vote/src/vote_state.rs

```diff
@@ -367,6 +367,22 @@ pub fn authorize(
     vote_account.set_state(&vote_state)
 }
 
+/// Update the node_pubkey, requires signature of the authorized voter
+pub fn update_node(
+    vote_account: &mut KeyedAccount,
+    node_pubkey: &Pubkey,
+    signers: &HashSet<Pubkey>,
+) -> Result<(), InstructionError> {
+    let mut vote_state: VoteState = vote_account.state()?;
+
+    // current authorized voter must say "yay"
+    verify_authorized_signer(&vote_state.authorized_voter, signers)?;
+
+    vote_state.node_pubkey = *node_pubkey;
```

Comment:
Just to make sure, is `bank.vote_accounts` will be updated in correct timing? So, immediately after this transaction is confirmed, the vote account should start voting from a validator with this new `node_pubkey` or next epoch??